server/routes/nl_interface.py

```diff
@@ -297,54 +302,104 @@ def _peer_buckets(sv2definition, svs_list):
   return peer_buckets
 
 
-@bp.route('/<path:place_dcid>')
-def page(place_dcid):
-  if os.environ.get('FLASK_ENV') == 'production':
-    flask.abort(404)
-  model = current_app.config['NL_MODEL']
-
-  # Step 1: find all relevant places and the name/type of the main place found.
-  # TODO(jehangiramjad): for now, using tmp_place_dcid. Replace with place detection.
-
-  place_types = dc.property_values([place_dcid], 'typeOf')[place_dcid]
-  main_place_type = _get_preferred_type(place_types)
-  main_place_name = dc.property_values([place_dcid], 'name')[place_dcid][0]
-
-  related_places = _related_places(place_dcid)
-  child_places_type = related_places['childPlacesType']
-  all_relevant_places = list(
-      set(related_places['parentPlaces'] + related_places['nearbyPlaces'] +
-          related_places['similarPlaces']))
-
-  # Step 2: replace the places in the query sentence with "".
-  # TODO(jehangiramjad): implement this. For now, using a temp query without a place.
-  query = request.args.get('q', 'people who cannot see')
+def _maps_place(place_str):
+  api_key = current_app.config["MAPS_API_KEY"]
+  url_formatted = f"{MAPS_API}input={place_str}&key={api_key}"
+  r = requests.get(url_formatted)
+  resp = r.json()
 
-  # Step 3: Identify the SV matched based on the query.
-  svs_df = pd.DataFrame(model.detect_svs(query))
+  # Return the first "political" place found.
+  if "results" in resp:
+    for res in resp["results"]:
+      if "political" in res["types"]:
+        return res
+  return {}
 
-  # Step 4: filter SVs based on scores.
-  highlight_svs = _highlight_svs(svs_df)
-  relevant_svs = _filtered_svs_list(svs_df)
+def _dc_recon(place_ids):
+  resp = dc.resolve_id(place_ids, "placeId", "dcid")
+  if "entities" not in resp:
+    return {}
 
-  # Step 5: get related SVGs and all info.
-  svgs_info = _related_svgs(relevant_svs, all_relevant_places)
+  d_return = {}
+  for ent in resp["entities"]:
+    for out in ent["outIds"]:
+      d_return[ent["inId"]] = out
+      break
 
-  # Step 6: get useful sv2name and sv2definitions.
-  sv_maps = _sv_definition_name_maps(svgs_info, relevant_svs)
-  sv2name = sv_maps["sv2name"]
-  sv2definition = sv_maps["sv2definition"]
+  return d_return
 
-  # Step 7: Get SVGs into peer buckets.
-  peer_buckets = _peer_buckets(sv2definition, relevant_svs)
+@bp.route('/')
+def page():
+  if os.environ.get('FLASK_ENV') == 'production':
+    flask.abort(404)
+  model = current_app.config['NL_MODEL']
+  query = request.args.get('q', 'people who cannot see')
```

Comment:
nit: just in case we want to override (for debugging, demo, or whatever reason), have a hidden dcid arg?  if that's set, we use that in place of place_dcid ?

Reply:
added the param **place_dcid**